athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/handlers/CompositeHandler.java

```diff
@@ -95,13 +95,22 @@ public final void handleRequest(InputStream inputStream, OutputStream outputStre
             throws IOException
     {
         try (BlockAllocatorImpl allocator = new BlockAllocatorImpl()) {
-            ObjectMapper objectMapper = VersionedObjectMapperFactory.create(allocator);
-            try (FederationRequest rawReq = objectMapper.readValue(inputStream, FederationRequest.class)) {
-                if (rawReq instanceof MetadataRequest) {
-                    ((MetadataRequest) rawReq).setContext(context);
-                }
-                handleRequest(allocator, rawReq, outputStream, objectMapper);
+            ObjectMapper objectMapper = VersionedObjectMapperFactory.create(allocator, SerDeVersion.SERDE_VERSION);
+            FederationRequest rawReq;
+            byte[] allInputBytes = inputStream.readAllBytes();
+
+            try {
+                rawReq = objectMapper.readValue(allInputBytes, FederationRequest.class);
+            }
+            catch (Exception e) { // if client has not upgraded to our latest, fallback to v2
```

Comment:
should we be catching all exceptions here? I'd think we'd only want to catch jackson related ones, no?

Reply:
Updated to IllegalStateException (the exception that throws in this case when I tested)